src/main/java/gregtech/api/cover/Cover.java

```diff
@@ -0,0 +1,255 @@
+package gregtech.api.cover;
+
+import codechicken.lib.raytracer.CuboidRayTraceResult;
+import codechicken.lib.render.CCRenderState;
+import codechicken.lib.render.pipeline.IVertexOperation;
+import codechicken.lib.vec.Cuboid6;
+import codechicken.lib.vec.Matrix4;
+import gregtech.client.utils.BloomEffectUtil;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.*;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.UnknownNullability;
+import org.jetbrains.annotations.Unmodifiable;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Consumer;
+
+public interface Cover {
+
+    /**
+     * @return the CoverableView containing this cover
+     */
+    @NotNull CoverableView getCoverable();
```

Comment:
This name feels a bit weird. I would instinctively want to call this `getCoverHolder`, but that name is already in use. Maybe just fully expanding this name to `getCoverableView`